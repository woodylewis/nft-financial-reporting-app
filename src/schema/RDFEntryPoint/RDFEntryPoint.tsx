import React, { Fragment, FunctionComponent } from 'react';
import PropTypes from 'prop-types';

import { EntryPoint } from 'schema-dts';
import { JsonLd } from 'react-schemaorg';

interface RDFEntryPointProps {
  id: string;
  appId: string;
}

const RDFEntryPoint: FunctionComponent<RDFEntryPointProps> = ({ id, appId }) => {
  const context = 'https://schema.org';
  const type = 'EntryPoint';
  const appType = 'SoftwareApplication';

  return (
    <Fragment>
      <JsonLd<EntryPoint>
        item={{
          '@context': context,
          '@id': id,
          '@type': type,
          actionApplication: {
            '@id': appId,
            '@type': appType,
          },
        }}
      />
    </Fragment>
  );
};

RDFEntryPoint.propTypes = {
  id: PropTypes.string.isRequired,
  appId: PropTypes.string.isRequired,
};

export default RDFEntryPoint;
