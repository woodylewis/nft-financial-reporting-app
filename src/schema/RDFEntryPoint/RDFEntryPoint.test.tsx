import React from 'react';
// import { shallow, ShallowWrapper, mount } from "enzyme";
import { create, ReactTestRenderer } from 'react-test-renderer';

// import { JsonLd } from "react-schemaorg";
import RDFEntryPoint from './RDFEntryPoint';

// let wrapper: ShallowWrapper;
let snapshot: ReactTestRenderer;
// let mountedInstance;

beforeEach(() => {
  const entryPoint = <RDFEntryPoint id='DSFSwitch' appId='SemanticSwitch' />;

  snapshot = create(entryPoint);
  // wrapper = shallow(entryPoint);
  // mountedInstance = mount(entryPoint);
});

describe('<RDFEntryPoint />', () => {
  test('it matches the snapshot', () => {
    expect(snapshot.toJSON()).toMatchSnapshot();
  });
});
