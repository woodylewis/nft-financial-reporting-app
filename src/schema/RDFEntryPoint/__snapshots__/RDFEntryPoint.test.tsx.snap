// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RDFEntryPoint /> it matches the snapshot 1`] = `
<script
  dangerouslySetInnerHTML={
    {
      "__html": "{"@context":"https://schema.org","@id":"DSFSwitch","@type":"EntryPoint","actionApplication":{"@id":"SemanticSwitch","@type":"SoftwareApplication"}}",
    }
  }
  type="application/ld+json"
/>
`;
